service/frontend/admin_handler.go

```diff
@@ -1656,7 +1656,11 @@ func (adh *AdminHandler) StreamWorkflowReplicationMessages(
 
 	shutdownChan := channel.NewShutdownOnce()
 	go func() {
-		defer shutdownChan.Shutdown()
+		defer func() {
+			shutdownChan.Shutdown()
+			err = serverCluster.CloseSend()
```

Comment:
Nit: Maybe wrap ctx on line 1651 with a cancel func and cancel the ctx after CloseSend()? Official grpc example seems doing that: https://github.com/grpc/grpc-go/blob/master/examples/route_guide/client/client.go#L120